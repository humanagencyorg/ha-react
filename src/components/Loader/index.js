import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import PulseLoader from 'react-spinners/PulseLoader';

const DEFAULT_LOADER_SIZE = 10;
const DEFAULT_LOADER_COLOR = '#61616a';
const StyledLoader = styled.div`
  position: absolute;
  z-index: 1;
  right: 1rem;
  top: 50%;
  transform: translate(0, -50%);
`;
function Loader({ loaderWrapper, loader }) {
  return (
    <StyledLoader {...loaderWrapper}>
      <PulseLoader
        size={DEFAULT_LOADER_SIZE}
        color={DEFAULT_LOADER_COLOR}
        {...loader}
      />
    </StyledLoader>
  );
}

Loader.propTypes = {
  loaderWrapper: PropTypes.object,
  loader: PropTypes.object,
};

export default Loader;
