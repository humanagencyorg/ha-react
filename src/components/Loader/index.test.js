import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Loader from './index';

describe('Loader component', () => {
  describe('when no props provided', () => {
    it('matches snapshot', () => {
      const component = renderer.create(
        <Loader />
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('when props provided', () => {
    it('matches snapshot', () => {
      const props = {
        loaderWrapper: {
          'data-testid': 'some fake test id',
          style: 'display: none;',
        },
      };

      const component = renderer.create(
        <Loader {...props} />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
