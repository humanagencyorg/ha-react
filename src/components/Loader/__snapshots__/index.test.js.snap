// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component when no props provided matches snapshot 1`] = `
.c0 {
  position: absolute;
  z-index: 1;
  right: 1rem;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

<div
  className="c0"
>
  <div
    className="css-0"
  >
    <div
      className="css-ybv0bb"
    />
    <div
      className="css-c51fk9"
    />
    <div
      className="css-uo8ju6"
    />
  </div>
</div>
`;

exports[`Loader component when props provided matches snapshot 1`] = `
.c0 {
  position: absolute;
  z-index: 1;
  right: 1rem;
  top: 50%;
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
}

<div
  className="c0"
  data-testid="some fake test id"
  style="display: none;"
>
  <div
    className="css-0"
  >
    <div
      className="css-ybv0bb"
    />
    <div
      className="css-c51fk9"
    />
    <div
      className="css-uo8ju6"
    />
  </div>
</div>
`;
